### PR TITLE
feature/order available

### PR DESCRIPTION
**Pull request by xidealo**

### CODE REVIEW

File: shared/src/commonTest/kotlin/com/bunbeauty/domain/feature/orderavailable/GetIsOrderAvailableUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    fun `invoke returns true when order is available`() = runTest {
+        // Given
+        val orderAvailable = orderAvailableMock.copy(available = true)
+        coEvery { orderAvailableRepository.getOrderAvailable() } returns orderAvailable
+
+        // When
+        val result = getOrderAvailableUseCase.invoke()
+
+        // Then
+        assertEquals(true, result)
```
Comment (MaxAstin) [inline shared/src/commonTest/kotlin/com/bunbeauty/domain/feature/orderavailable/GetIsOrderAvailableUseCase.kt:39]: `assertTrue`

File: shared/src/iosMain/kotlin/com/bunbeauty/shared/data/DataStoreRepository.kt [hunk 2]
```diff
@@ -6,10 +6,12 @@ import com.bunbeauty.shared.domain.model.Discount
 import com.bunbeauty.shared.domain.model.Payment
 import com.bunbeauty.shared.domain.model.Settings
 import com.bunbeauty.shared.domain.model.UserCityUuid
+import com.bunbeauty.shared.domain.model.order.OrderAvailable
```
Comment (MaxAstin) [inline shared/src/iosMain/kotlin/com/bunbeauty/shared/data/DataStoreRepository.kt:9]: extra import

**Review by MaxAstin — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt [hunk 3]
```diff
@@ -136,6 +138,29 @@ class MainActivity : AppCompatActivity(R.layout.layout_compose), IMessageHost {
...
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .background(FoodDeliveryTheme.colors.statusColors.warning)
+                    .padding(8.dp),
+                contentAlignment = Alignment.Center
+            ) {
+                Text(
+                    text = resources.getString(R.string.warning_no_internet),
+                    style = FoodDeliveryTheme.typography.bodyMedium,
+                    color = FoodDeliveryTheme.colors.mainColors.onError
```
[the remark this answers is not in the document]
Comment (xidealo) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt:158]: fix

File: app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt [hunk 4]
```diff
@@ -136,6 +138,29 @@ class MainActivity : AppCompatActivity(R.layout.layout_compose), IMessageHost {
...
+        ) {
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .background(FoodDeliveryTheme.colors.statusColors.warning)
+                    .padding(8.dp),
+                contentAlignment = Alignment.Center
+            ) {
+                Text(
+                    text = resources.getString(R.string.warning_no_internet),
```
[the remark this answers is not in the document]
Comment (xidealo) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt:156]: fix

File: app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt [hunk 5]
```diff
@@ -136,6 +138,29 @@ class MainActivity : AppCompatActivity(R.layout.layout_compose), IMessageHost {
         }
     }
 
+    @Composable
+    private fun OrderAvailableMessage(visible: Boolean) {
+        AnimatedVisibility(
+            visible = visible,
+            enter = fadeIn(tween(300)),
+            exit = fadeOut(tween(300))
```
[the remark this answers is not in the document]
Comment (xidealo) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt:146]: fix

File: shared/src/commonTest/kotlin/com/bunbeauty/domain/feature/orderavailable/GetIsOrderAvailableUseCase.kt (hunk 1 above)
Comment (xidealo) [reply shared/src/commonTest/kotlin/com/bunbeauty/domain/feature/orderavailable/GetIsOrderAvailableUseCase.kt:39]: +

File: app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (xidealo) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt:146]: foiix

File: app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (xidealo) [reply app/src/main/java/com/bunbeauty/papakarlo/feature/main/MainActivity.kt:156]: ix

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/orderavailable/GetOrderAvailableUseCase.kt [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.bunbeauty.shared.domain.feature.orderavailable
+
+import com.bunbeauty.shared.data.repository.OrderAvailableRepository
+
+private const val ORDER_AVAILABLE = true
+
+class GetOrderAvailableUseCase(
```
[the remark this answers is not in the document]
Comment (xidealo) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/orderavailable/GetOrderAvailableUseCase.kt:7]: fixc

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/orderavailable/GetOrderAvailableUseCase.kt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package com.bunbeauty.shared.domain.feature.orderavailable
+
+import com.bunbeauty.shared.data.repository.OrderAvailableRepository
+
+private const val ORDER_AVAILABLE = true
```
[the remark this answers is not in the document]
Comment (xidealo) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/orderavailable/GetOrderAvailableUseCase.kt:5]: fix

File: shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/model/order/OrderAvailable.kt [hunk 8]
```diff
@@ -0,0 +1,5 @@
+package com.bunbeauty.shared.domain.model.order
+
+data class OrderAvailable(
```
[the remark this answers is not in the document]
Comment (xidealo) [reply shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/model/order/OrderAvailable.kt:3]: fix

**Review by MaxAstin — APPROVED** [review]

**Pull request merged**